packages/block-editor/src/components/list-view/block-contents.js

```diff
@@ -7,16 +7,25 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { useSelect } from '@wordpress/data';
-import { forwardRef } from '@wordpress/element';
+import { forwardRef, useEffect, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import { unlock } from '../../lock-unlock';
 import ListViewBlockSelectButton from './block-select-button';
 import BlockDraggable from '../block-draggable';
 import { store as blockEditorStore } from '../../store';
+import { updateAttributes } from './update-attributes';
+import { LinkUI } from './link-ui';
+import { useInsertedBlock } from './use-inserted-block';
 import { useListViewContext } from './context';
 
+const BLOCKS_WITH_LINK_UI_SUPPORT = [
```

Comment:
Similar feedback to this comment: https://github.com/WordPress/gutenberg/pull/49417#discussion_r1189115306 — is this change baking in too much knowledge about particular blocks into the `ListView` component?

Would it be possible to split out the link UI behaviour so that it's added to the list view by the consumer of the component, a bit like how there's now the `renderAdditionalBlockUI` prop as of https://github.com/WordPress/gutenberg/pull/50465?